src/android/FirebaseMessagingPlugin.java

```diff
@@ -137,7 +142,14 @@ private void requestPermission(CordovaArgs args, CallbackContext callbackContext
         if (NotificationManagerCompat.from(context).areNotificationsEnabled()) {
             callbackContext.success();
         } else {
-            callbackContext.error("Notifications permission is not granted");
+            if (Build.VERSION.SDK_INT >= 32) {
+                if (ContextCompat.checkSelfPermission(null,
```

Comment:
Use `cordova.hasPermission(Manifest.permission.POST_NOTIFICATIONS)` to verify if permission is already granted for cordova framework